text/0000-modular-crypto-service.md

```diff
@@ -79,29 +79,170 @@ explain more fully how the detailed proposal makes those examples work.
 [**WIP**] along with previous feature examples
 ## Detail proposal for BCCSP
 ```
-- sw(adjust)
-- sm2(sample for new crypto standard)
-- util(adjust/new added as class loader)
+- sw(adjust/new added as class loader)
+```
+Overall speaking, it seems we need to implement a crypto wrapper in bccsp package.
+Common parts:
+In BCCSP we need to:
+- Refactor current bccsp with new process logic.
+- Impl for new crypto standard with api lists.
+
+#### Here is the logic changed in current sw to fill with wrapper above.
+```go
+package sw
+
+publicKeyToEncryptedPEM(k interface{}, pwd []byte){
+	for i,v := range GetfromWrapperMap() {
+		if k.type == i.type {
+			return do(k,v)
+		}
+	}
+	return "not supported"
+}
 ```
-Overall speaking, it seems we are going to impl a classloader logic for bccsp.
-- refactor current bccsp with new process logic.
-- impl for new crypto standard with api lists.
 
-### Refactor current bccsp with new process logic.
-To make something as classloader in bccsp. There may need to build some global level map in `<type, function>` way. For example with below logic, we can reused `fileks.go` and by reflect, the code will auto switch between crypto logics which added.
+#### A sample function as specific crypto wrapper/provider
+```go
+package specific
+import specific_crypto_lib
+
+//example function
+publicKeyToEncryptedPEM(k interface{}, pwd []byte){
+    specific impl
+}
+
 ```
-		// TWGC todo
-		// make this map as global variable
-		var m map[reflect.Type]func(interface{}) bccsp.Key
-		m = make(map[reflect.Type]func(interface{}) bccsp.Key)
-		m[reflect.TypeOf(&ecdsa.PrivateKey{})] = NewECDSAPrivateKey
-		m[reflect.TypeOf(&sm2.PrivateKey{})] = NewSM2PrivateKey
-		for i, v := range m {
-			if i == reflect.TypeOf(key) {
-				return v(key), nil
-			}
-		}
+
+#### Here is logic need to be added as a global wrapper to load plugin
+currently, some part of logic is implemented here. https://github.com/Hyperledger-TWGC/fabric/tree/bccsp-gm
+```go
```

Comment:
Link to code lines than long raw code block in proposal is preferred

Reply:
I am agree with this one, but how we line to a raw code when the code still in PR?

Reply:
No matter PR is created or not, the code is always in a branch somewhere, use `copy permalink` to get it. 
![image](https://user-images.githubusercontent.com/7227589/144065169-577547ec-086a-4e4b-8156-e73b39217b75.png)

Reply:
copy